rocAL/rocAL/source/augmentations/effects_augmentations/node_snp_noise.cpp

```diff
@@ -21,42 +21,56 @@ THE SOFTWARE.
 */
 
 #include <vx_ext_rpp.h>
-#include <VX/vx_compatibility.h>
 #include "node_snp_noise.h"
 #include "exception.h"
 
 
 SnPNoiseNode::SnPNoiseNode(const std::vector<Tensor *> &inputs, const std::vector<Tensor *> &outputs) :
         Node(inputs, outputs),
-        _sdev(SDEV_RANGE[0], SDEV_RANGE[1])
+        _noise_prob(NOISE_PROB_RANGE[0], NOISE_PROB_RANGE[1]),
+        _salt_prob (SALT_PROB_RANGE[0], SALT_PROB_RANGE[1]),
+        _noise_value(NOISE_RANGE[0], NOISE_RANGE[1]),
+        _salt_value(SALT_RANGE[0], SALT_RANGE[1])
 {
 }
 
-void SnPNoiseNode::create_node()
-{
+void SnPNoiseNode::create_node() {
     if(_node)
         return;
 
-    _sdev.create_array(_graph , VX_TYPE_FLOAT32, _batch_size);
-    // _node = vxExtrppNode_NoisebatchPD(_graph->get(), _inputs[0]->handle(), _src_roi_width, _src_roi_height, _outputs[0]->handle(), _sdev.default_array(), _batch_size);
+    _noise_prob.create_array(_graph, VX_TYPE_FLOAT32, _batch_size);
+    _salt_prob.create_array(_graph, VX_TYPE_FLOAT32, _batch_size);
+    _noise_value.create_array(_graph, VX_TYPE_FLOAT32, _batch_size);
+    _salt_value.create_array(_graph, VX_TYPE_FLOAT32, _batch_size);
+    vx_scalar seed = vxCreateScalar(vxGetContext((vx_reference)_graph->get()), VX_TYPE_UINT32, &_seed);
+
+    _node = vxRppNoise(_graph->get(), _inputs[0]->handle(), _src_tensor_roi, _outputs[0]->handle(),_noise_prob.default_array(), _salt_prob.default_array(), _noise_value.default_array(), _salt_value.default_array(), seed, _input_layout, _output_layout, _roi_type);
```

Comment:
Why this node has been named as vxRppNoise? it should be ideally something like vxRppSNPNoise
There are shot noise noise and gaussian also available in RPP, so would suggest to differentiate here

Reply:
Will make the change when we add support for other noise. Please note